File: ruby/silly_years.rb
def silly_years(year)
  years = []
  curr_year = year + 1
  while years.length < 10
    year_s = curr_year.to_s

    left, right, mid = [year_s[0..1], year_s[2..3], year_s[1..2]].map(&:to_i)

    years << curr_year if left + right == mid

    curr_year += 1
  end

  years
end

p silly_years(1989)

File: js/aa/sillyYears.js
function sillyYears(year) {
  const years = [];
  let currYear = year + 1;
  while (years.length < 10) {
    if (isSilly(currYear)) {
      years.push(currYear);
    }
    currYear++
  }
  return years;
}

function isSilly(year) {
  const yearS = year.toString();
  const left = yearS.slice(0, 2);
  const right = yearS.slice(2);
  const mid = yearS.slice(1,3);
  return parseInt(left) + parseInt(right) === parseInt(mid);
}

console.log(sillyYears(1989));
